model_analyzer/config/input/objects/config_model_analysis_spec.py

```diff
@@ -55,6 +62,16 @@ def constraints(self):
 
         return self._constraints
 
+    def weightings(self):
```

Comment:
I can't tell where weights are a single value, and where they are a list. Type checking would help, as would proper pluralization. The name implies it returns multiple weights, but I think this is just a scalar int?

Also, can we populate this with default values so that we don't have to handle "or none" all over the place?

Reply:
Yes, this should not be plural. I went to add type checking and then got a ton of complaints because not all the variables in the methods were type checked. I tried to boyscout the change, but it was too onerous.